amplify/backend/api/colonycdapp/schema.graphql

```diff
@@ -501,4 +513,99 @@ type ColonyAction @model {
   toDomain: Domain @hasOne(fields: ["toDomainId"])
   fundamentalChainId: Int
   newColonyVersion: Int
+  roles: ColonyActionRoles
+
+  # Required since some actions might have multiple event entries in the list
+  # which need to be iterated over and exposed individually
+  #
+  # This should be a list/array, but this is yet another thing Amplify is bad at
+  # The official response from them is to use DynamoDB directly:
+  # https://stackoverflow.com/questions/50992106/append-item-to-list-using-aws-appsync-to-dynamodb
+  # Which is just freaking great!
+  #
+  # As such, I've decided to simplify this, an just pass a stringified JSON around
+  # since this is for display purpouses only.
+  #
+  # If anyone has a better idea on how to handle this, I'll gladly change it!
+  #
+  # When / if you decide to use it, just be mindful of what format other people
+  # are using, and follow suit.
+  #
+  # Roles Types:
+  # [{ id: String, type: String, role: Number, setTo: boolean }]
+  individualEvents: String
+}
+
+type ColonyActionRoles {
+  role_0: Boolean # recovery
+  role_1: Boolean # root
+  role_2: Boolean # arbitration
+  role_3: Boolean # architecture
+  role_5: Boolean # funding
+  role_6: Boolean # administration
+}
+
+# Roles have been set as role_<number> so we can more easily map them to their
+# actual contract counterpart on the client and block ingestor side(s)
+
+type ColonyRole @model {
+  id: ID! @index # colonyAddress_domainNativeId_userAddress_roles
+  domain: Domain! @hasOne
```

Comment:
Nitpick: it's not clear which of these params are variable and constant in this comment. 

E.g. would be clearer:
``` graphql
# <colonyAddress>_<domainNativeId>_<contractAddress>_roles
```

Reply:
Also, is there a reason for marking this as:

`id: ID! @index` ?

It's the primary key by default so I'm not sure what we gain from adding the `@index`